app/controllers/posts_controller.rb

```diff
@@ -4,6 +4,35 @@ def index
     @posts = User.find(params[:user_id]).posts
   end
 
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: } }
+    end
+  end
+
+  def create
+    # new object from params
+    post = Post.new(title: params[:new_post][:title], text: params[:new_post][:text], user: current_user)
+    # respond_to block
+    respond_to do |format|
+      format.html do
+        # if question saves
+        if post.save
+          # success message
+          flash[:notice] = 'New post saved successfully'
+          # redirect to index
+          redirect_to user_posts_path(current_user.id)
+        else
+          # error message
+          flash.now[:notice] = "Error: Couldn't create post"
+          # render new
+          render :new, locals: { post: }
+        end
+      end
+    end
+  end
+
```

Comment:
Great job @CindyDorantes 🤝

- [ ] [Optional] You may also add a params method to the posts_controller. You can learn more about strong parameters by following this [link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).

Reply:
sure! Thanks for the comment